Description: Add a target for docu-generation and remove this from ALL
 This patch adds a target that makes is possibe to create the Doxygen
 docu from the base build directory without invoking the ALL target.
 At the same time this documentation creation is removed from the ALL
 target to avoid the it is build various times. Rationale is to enable
 binary-arch/indep and install-arch/indep without the overhead of
 recreating the documentation various times. (Unfortunately the make
 seems to re-run at least the PDF document creation each time.
 (This is a Debian specific patch). 
Author: Gert Wollny <gw.fossdev@gmail.com>
Forwarded: not-needed 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -554,6 +554,12 @@
 endif()
 #-----------------------------------------------------------------------------
 
+# Add a target so that the documentation can be build from the
+# root dir
+if(GDCM_DOCUMENTATION)
+  add_custom_target(DoxygenDoc)
+endif()
+
 # Need to subdirs in Source/Common before Wrapping
 # to have gdcmConfigure.h around
 subdirs(Utilities)
--- a/Utilities/doxygen/CMakeLists.txt
+++ b/Utilities/doxygen/CMakeLists.txt
@@ -154,7 +154,6 @@
     endif()
     # add target to 'ALL'
     add_custom_target(GDCMDoxygenPDF
-      ALL
       DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/latex/gdcm-${GDCM_VERSION}.pdf
       COMMENT  "GDCM: Executing GDCMDoxygenPDF"
     )
@@ -162,13 +161,14 @@
       ${CMAKE_CURRENT_BINARY_DIR}/latex/gdcm-${GDCM_VERSION}.pdf
       DESTINATION ${GDCM_INSTALL_DOC_DIR}
     )
+    add_dependencies(DoxygenDoc GDCMDoxygenPDF)
   else()
     # make DoxygenDoc depends on the final tarball thus all file are garantee to be generated
     add_custom_target(GDCMDoxygenDoc
-      ALL
       DEPENDS ${GDCM_DOC_TARBALL}
       COMMENT  "GDCM: Executing GDCMDoxygenDoc"
     )
+    add_dependency(DoxygenDoc GDCMDoxygenDoc)
   endif()
 
   # Install html pages:
--- a/Utilities/doxygen/vtk/CMakeLists.txt
+++ b/Utilities/doxygen/vtk/CMakeLists.txt
@@ -103,7 +103,6 @@
     ${CMAKE_CURRENT_BINARY_DIR}/doc_makeall.sh @ONLY )
 
   add_custom_target(${DOXYGEN_PROJECT_NAME}DoxygenDoc
-    ALL
     ${BASH}
     ${CMAKE_CURRENT_BINARY_DIR}/doc_makeall.sh)
 
@@ -111,6 +110,8 @@
     ${DOXYGEN_PROJECT_NAME}DoxygenDoc
     GDCMDoxygenPDF
   )
+  add_dependencies(DoxygenDoc
+    ${DOXYGEN_PROJECT_NAME}DoxygenDoc)
 
   if(INCLUDE_QT_DOCUMENTATION)
     add_custom_target(QtDoxygenTags
